server/testdriver/core/src/main/java/org/infinispan/server/test/core/rollingupgrade/RollingUpgradeHandler.java

```diff
@@ -385,7 +382,8 @@ private ConfigAndDriver startNode(boolean toOrFrom, int nodeCount, int expectedC
          builder.property(TestSystemPropertyNames.INFINISPAN_TEST_SERVER_CONTAINER_VOLUME_REQUIRED, "true");
       }
 
-      String versionToUse = toOrFrom ? configuration.toVersion() : configuration.fromVersion();
+      String prefix = (toOrFrom ? "To-" : "From-") + clusterName;
+      String versionToUse = prefix + (toOrFrom ? configuration.toVersion() : configuration.fromVersion());
 
       if (versionToUse.startsWith("image://")) {
```

Comment:
Looks like the if/else here will always be false? Since the versionToUse will always start with To- or From-.